src/entities/activity/lib/hooks/useSupportableActivities.ts

```diff
@@ -1,26 +1,15 @@
-import { useMemo } from "react"
-
 import { useActivityByIdQuery } from "../../api"
-import { activityBuilder } from "../../model"
 
 type Props = {
   isPublic: boolean
   activityId: string
 }
 
-export const useSupportableActivity = (props: Props) => {
-  const { data, isError, isLoading } = useActivityByIdQuery(props)
-
-  const activity = useMemo(() => {
-    return data?.data?.result
-  }, [data])
-
-  const isSupportedActivity = useMemo(() => {
-    return activityBuilder.isSupportedActivity(activity)
-  }, [activity])
+export const useActivity = (props: Props) => {
+  const { data: activity, isError, isLoading } = useActivityByIdQuery(props, { select: data => data.data.result })
```

Comment:
Please create a mapper for Activity and use it in the select callback

Reply:
Done